scripts/set_gem_timing

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+NC_STRING="NC"
+SC_STRING="SC"
+# TODO:(josh) reflect on absolute paths
+IOC_PATH_NC="/reg/g/pcds/epics/ioc/fee/GasDetDAQ/R4.0.22"
+IOC_PATH_SC="/reg/g/pcds/epics/ioc/fee/GasDetDAQ/R4.0.22-NOTS"
+IOC_PATH=0
+
+# User input validation
+if [[ -n $1 ]]; then # if the user passed at least one arg
+  if [[ $1 != $NC_STRING ]] && [[ $1 != $SC_STRING ]]; then #validate argument is either SC or NC
+    echo "Error. Argument must be either: [NC or SC]"
+    exit 1
+  else
+    echo "User passed argument of: $1"
+  fi
+else
+  echo "Error. Must pass at least one argument"
+  exit 1
+fi
+
+if [[ -n $2 ]]; then
+  echo "Warning. Must pass strictly one argument, additional arg: $2 is ignored"
+  exit 1
+fi
+
+# check current operational state
+OP_MODE=$(caget KFE:CAM:TPR:02:MODE | awk '{print $2}')
+SET_MODE_CMD="caput KFE:CAM:TPR:02:MODE $1"
+
+echo "Current OP_MODE is: $OP_MODE"
+if [[ $OP_MODE = $1 ]]; then
+  echo "Already in requested mode: $OP_MODE exitting with NOP"
+  exit 1
+else
+  echo "Actuating requested timing change, caputing $1 to KFE:CAM:TPR:02:MODE"
+  $SET_MODE_CMD
+fi
+
+# only now discern which mode we want to put it in, TODO(josh): theres some more bashy way of string unwrapping todo this....
+if [[ $1 = $NC_STRING ]]; then
+  IOC_PATH=$IOC_PATH_NC
+else
+  IOC_PATH=$IOC_PATH_SC
+fi
+
+SET_IOC_PATH_CMD="imgr ioc-lfe-gasdet-daq --upgrade $IOC_PATH"
```

Comment:
Same question as above: why is this broken out as a string prior to executing it?

Reply:
I'll respond to both here. It's been a good few years since I've written any bash but I generally (especially in a low to no production gate system like we have here) will set my intended command to a string variable so I can echo it in testing as opposed to running. When I was cleaning up I somewhat intentionally left this artifact, if it obfuscates the code I am more than happy to remove. What do you think?

Reply:
Gotcha, that makes sense. There's no need to remove it, I was simply curious.